controllers/wavefrontalert_controller.go

```diff
@@ -245,20 +246,39 @@ func (r *WavefrontAlertReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			respAlert.State = state
 			respAlert.LastChangeChecksum = a.LastChangeChecksum
 			// if even one of the child got failed, make parent status as error
-			status.State = state
-			status.RetryCount = wfAlert.Status.RetryCount + 1
+			wfAlert.Status.State = state
+			wfAlert.Status.RetryCount = wfAlert.Status.RetryCount + 1
 		}
 		log.Info("alert ids before and after", "before", a.ID, "after", alert.ID)
 		respAlert.State = alertmanagerv1alpha1.Ready
-		respAlerts = append(respAlerts, respAlert)
+		//TODO: Figure out a better way to handle this in future when we have multiple
+		wfAlert.Status.State = alertmanagerv1alpha1.Ready
```

Comment:
If the alert was failed to update you already updated the ```wfAlert.Status.State``` to error in line 249, and here you overwrite it again? same with line 253 ```respAlert.State = alertmanagerv1alpha1.Ready``` and line 246 ```espAlert.State = state```

Reply:
You are right- Its supposed to be state value which was set during the inside loop. There is definitely more thought needs to be done before we release initial version on this case if there is a better way to handle this